### ISSUE TITLE
move farther away from the main/overlays repos notion

### ISSUE THREAD

**Issue by radhermit** [pkgcore/pkgcore#65]
Basically all repos should have priorities/weights assigned to them and be iterated through in that order. This should mostly possible in the current scheme, the relevant code just needs to be updated and documented as working in that fashion.

Also, the default verbose output probably should show always show the repo_id even for the repo with the highest priority (current main repo).

**Comment by radhermit** [comment]
Mostly done on the config side, some work remains to properly handle multiple masters on the repo side.

**Comment by reith** [comment]
I see priorities as put in `repos.conf` being not handled in the same way portage doing.  In portage higher priorities win and priority of default Gentoo repo is -1000.

**Comment by radhermit** [comment]
Isn't the only difference being that pkgcore currently uses -9999? I'm not sure how that's an issue for you unless you're using priorities between -1000 and -9999 for repos.

**Comment by reith** [comment]
Yeah, that was it. Thanks.  I set explicitly priority of Gentoo repo to make portage and pkgcore behave same.

**Comment by radhermit** [comment]
I've just synced pkgcore's default repo priority with portage's. So it shouldn't be an issue going forward.
